app/models/topic_converter.rb

```diff
@@ -52,13 +39,24 @@ def convert_to_private_message
       @topic.archetype = Archetype.private_message
       add_allowed_users
       @topic.save!
+      update_user_actions
       watch_topic(topic)
     end
     @topic
   end
 
   private
 
+  def update_user_actions
+    UserAction.where(
+      target_topic_id: @topic.id,
+      action_type: [UserAction::GOT_PRIVATE_MESSAGE, UserAction::NEW_PRIVATE_MESSAGE]).find_each do |ua|
+        UserAction.remove_action!(ua.attributes.symbolize_keys.slice(:action_type, :user_id, :acting_user_id, :target_topic_id, :target_post_id))
+      end
+    @topic.posts.each { |post| UserActionManager.post_created(post) }
```

Comment:
This is a good point, I will make a note about this and will follow up. Topic Conversion was already pretty slow, but on a large topic this could be a time bomb.